### PR TITLE
add defaulting for customresources

### PR DESCRIPTION
**Pull request by deads2k**
This adds the promised defaulting for customresources.  Namespaced by default, listkind=kind+List, singular=toLower(kind).

### CODE REVIEW

**Comment by deads2k** [comment]
@k8s-bot unit test this

File: staging/src/k8s.io/kube-apiextensions-server/pkg/apis/apiextensions/install/install_test.go [hunk 1]
```diff
@@ -17,11 +17,36 @@ limitations under the License.
...
 
+	"github.com/google/gofuzz"
+
 	apitesting "k8s.io/apimachinery/pkg/api/testing"
+	"k8s.io/kube-apiextensions-server/pkg/apis/apiextensions"
 )
 
 func TestRoundTripTypes(t *testing.T) {
-	apitesting.RoundTripTestForAPIGroup(t, Install, nil)
+	apitesting.RoundTripTestForAPIGroup(t, Install, apitesting.MergeFuzzerFuncs(t,
```
Comment (sttts) [inline staging/src/k8s.io/kube-apiextensions-server/pkg/apis/apiextensions/install/install_test.go]: nit: no need to merge only one set of funcs

**Comment by deads2k** [comment]
its the api-extensions thing again: #45490 approving

**Pull request merged**